### ISSUE TITLE
Several issues about ore mining and melting in 22w38a

### ISSUE THREAD

**Issue by cyberjasse** [tsunamayo/Starship-EVO#4922]
### Short Description of the issue:
i encountered several issues when trying to mine ore and melting them in order to test the 22w38a.

I created a new world and went to the habitat. I switched from the habitat and a ship near asteroids to mine ore then return on habitat to melt and craft.

- There are stones that we cannot combine and I don't know why
![cannotCombineStone](https://user-images.githubusercontent.com/15956751/191599237-c1155bcb-a62a-4900-9bba-34c4e5f0fff4.png)
- We can fit more ores in a container that what it can contains if the ore is the same
![canFitMoreOre](https://user-images.githubusercontent.com/15956751/191599508-064554fe-0da1-45e8-bd6f-0488e1215bf9.png)
- If we open the furnace menu and do a shift clic on an ore in our inventory, it dissapears
- I put some ore in a furnace and nothing came out but i did not placed a treadmill in front of the output
- Sometime when putting several ingots in a furnace, it outputs ingots and ores that cannot be taken.
- The ore pass through the ground in the habitat
- When i launch 2 ores into the furnace, The first output gives a ingot that does not disappear when I take it one time but I can take it several times, between 10 and 100) before it despawned in the world. So it gaves me several ingot in my inventory. I had one time 93 ingots with 1 single ore. The second entered ore outputs a single ingots.
- Talking about that, when i mine, An ore can be taken several times like this and thus I have easily lot of ores. I have lot of ore too when I let the collector collect them. There are more ore in the collector than ore that came out of what i mined. I wonder if it was inteded. It was a large mining beam.
- Some hull bloc and a big container dissapeared graphicaly but still there.
- Then I exited to title and reloaded the world and I have again the bug with infinite loading time.

### Environment
- Starship EVO version: 22w38a experimental
- Graphics Card: RTX2070 super

### Additional information:
the playerlog:
[Player.log](https://github.com/tsunamayo/Starship-EVO/files/9620052/Player.log)

the save:
[experimentalCraftWorldTest.zip](https://github.com/tsunamayo/Starship-EVO/files/9620056/experimentalCraftWorldTest.zip)

**Comment by cyberjasse** [comment]
I have tested the assembler a bit and found no issue about it.
The infinite loading time on a world was encountered in 22w33a : https://github.com/tsunamayo/Starship-EVO/issues/4901

**Comment by tsunamayo** [comment]
@cyberjasse Hi, thanks for all the issues reported. I will come back to you asap! Cheers

**Comment by tsunamayo** [comment]
@cyberjasse Hi,
So regarding the following issues:

- There are stones that we cannot combine and I don't know why => fixed for next patch

- If we open the furnace menu and do a shift clic on an ore in our inventory, it dissapears => SHIFT click to clear a slot is a feature. I fixed the issue that no item was dropped (it is still not fixed for bricks though, as you cant drop bricks atm)

- I put some ore in a furnace and nothing came out but i did not placed a treadmill in front of the output => you need to toggle the option inside the Advanced Setting of the furnace.

- Sometime when putting several ingots in a furnace, it outputs ingots and ores that cannot be taken. => I cant reproduce that issue, do you have any clue on what step to reproduce?

- The ore pass through the ground in the habitat => this is "normal", the habitat terrain needs to be migrated to the new technology so it will be for another patch.

- When i launch 2 ores into the furnace, The first output gives a ingot that does not disappear when I take it one time but I can take it several times, between 10 and 100) before it despawned in the world. So it gaves me several ingot in my inventory. I had one time 93 ingots with 1 single ore. The second entered ore outputs a single ingots. => not sure to understand the test case. Right now "lauching" an item like an ore duplicate it (it is convenient for  testing), but of course for survival I will remove that.

- Talking about that, when i mine, An ore can be taken several times like this and thus I have easily lot of ores. I have lot of ore too when I let the collector collect them. There are more ore in the collector than ore that came out of what i mined. I wonder if it was inteded. It was a large mining beam. => I will try to check this.

- Some hull bloc and a big container dissapeared graphicaly but still there. => I will need a proper test case to investigate this. How did it happened

- Then I exited to title and reloaded the world and I have again the bug with infinite loading time.
=> I am loading that save just fine. But I can see the log file issue. Are you sure you quit to windows before trying to reload? When there is an issue you always need to do that.

Thanks a lot for the big list!

**Comment by cyberjasse** [comment]
Hello @tsunamayo . Thank you for your response .

Now I am in version 22w38b.
I tried to reproduce the bug when we enter several ingots in the furnace. This time the only difference between this attempt and previous is that the outputed ore gives me infinite ingots.  it does not happen if I enter only one ingot and wait until the melt is finished. In this video, the furnace output ore in addition to ingots and ingots are infinite ingots : 

https://user-images.githubusercontent.com/15956751/191829225-9fc0b5e6-aeda-491e-85a1-f1f65b50e158.mp4

The playerlog of this attempt  : 
[outputsOresAndBonusIngotsPlayerlogs.log](https://github.com/tsunamayo/Starship-EVO/files/9628665/outputsOresAndBonusIngotsPlayerlogs.log)

I also recorded that mined ore can be taken several times : 

https://user-images.githubusercontent.com/15956751/191829864-bd6de4b1-67cf-4c43-bb6e-9ecbd3d86365.mp4

About the infinite loading time, you were right, I exited to title and not to windows. I can open my save now. This evening, I opened a save then exited to title directly after the save loaded and then tried to open the save that you tried and I have again infinite loading time. It seems to be a problem when we exit to title. The logs looks like an infinite loop of Null reference : 

[exitToTitleThenLoadPlayerlog.log](https://github.com/tsunamayo/Starship-EVO/files/9628718/exitToTitleThenLoadPlayerlog.log)

**Comment by tsunamayo** [comment]
@cyberjasse Hi, so to follow-up on this I cant reproduce the furnace and the quit to title issue. I did a few fixes a few day ago it might have fixed it - could you retry in 22w41a? Thanks!

**Comment by cyberjasse** [comment]
Hello @tsunamayo . Thank you for your bugfixes. I retried in 22w41a and yes these issues are fixed.

During the test, I launched a lot of silver ore to the furnace and one of the produced ingot gave me the "infinite ingot" bug that I described once here. I mean that if I grab it, it gives an ingot in my inventory but does not despawn in the world, allowing me to re-grab it to infinity. Unfortunately I still not found a way to reproduce it. 

There still also 2 issues ;
- We can fit more ore in a container than its capacity as long as its the same ore
- After mining an asteroid, we can often grab ore several times (the video OreTakenSeveralTimes.mp4 above)
